jwst/tweakreg/wcsimage.py

```diff
@@ -209,51 +211,41 @@ def set_correction(self, matrix=[[1, 0], [0, 1]], shift=[0, 0]):
         self._update_transformations()
 
     def _check_wcs_structure(self, wcs):
-        valid = True
-        message = ""
         if wcs is None or wcs.pipeline is None:
-            message = "WCS is None."
-            valid = False
-            return valid, message
+            return False, "Either WCS or its pipeline is None."
 
         frms = wcs.available_frames
         nframes = len(frms)
+
         if nframes < 3:
-            message = "There are fewer than 3 frames in the WCS pipeline."
-            valid = False
-            return valid, message
+            return False, "There are fewer than 3 frames in the WCS pipeline."
 
         if frms.count(frms[0]) > 1 or frms.count(frms[-1]) > 1:
-            valid = False
-            message = "One or more frames in the WCS pipeline are not unique."
-            return valid, message
+            return (False, "First and last frames in the WCS pipeline must "
+                    "be unique.")
```

Comment:
BTW, all frame names in the pipeline **must** be unique.

Reply:
@nden Is this guaranteed? If yes, then I can remove this `IF`. However, when using the standard example from the `gWCS` docs modified to have two identical frames, it does not raise an exception: `wcsobj = wcs.WCS([('det', det2sky), ('det', det2sky), ('sky', None)])` (but the code does crash when attempting to evaluate forward transform.